src/parse.ts

```diff
@@ -20,11 +20,11 @@ export interface ParsedHost {
 }
 
 export function parseURL (input: string = ''): ParsedURL {
-  if (!hasProtocol(input)) {
+  if (!isURL(input, true)) {
     return parsePath(input)
   }
 
-  const [protocol, auth, hostAndPath] = (input.match(/([^:/]+:)\/\/([^/@]+@)?(.*)/) || []).splice(1)
+  const [protocol = '', auth, hostAndPath] = (input.match(/([^:/]+:)?\/\/([^/@]+@)?(.*)/) || []).splice(1)
```

Comment:
`//` is not really a valid URL even considering scheme is optional since has no path:

```
URI = scheme:[//authority]path[?query][#fragment]
```

Maybe we can try this: 

```suggestion
  const [protocol = '', auth, hostAndPath] = (input.match(/([^:/]+:)?\/\/([^/@]+@)?(.+)/) || []).splice(1)
```